pandas/core/frame.py

```diff
@@ -3821,13 +3821,16 @@ def _apply_standard(self, func, axis, ignore_failures=False, reduce=True):
         # e.g. if we want to apply to a SparseFrame, then can't directly reduce
         if reduce:
 
-            try:
-
-                # the is the fast-path
-                values = self.values
-                dummy = Series(NA, index=self._get_axis(axis),
-                               dtype=values.dtype)
+            # the is the fast-path
+            values = self.values
+            # Create a dummy Series from an empty array
+            # Unlike filling with NA, this works for any dtype
+            index = self._get_axis(axis)
+            empty_arr = np.empty(len(index), dtype=values.dtype)
+            dummy = Series(empty_arr, index=self._get_axis(axis),
+                           dtype=values.dtype)
```

Comment:
I can't think of a test that will isolate this change, as the code is wrapped in DF._apply_standard, and the dummy array isn't returned. Part of the previous issue was that all exceptions were caught, so when it failed to create the dummy array, this branch silently failed. I took the dummy generation code outside of the try block, so if it fails, it will raise an exception in the one test I added. Otherwise, I'm not sure what else I can do to test it.